clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,422 @@
+    for bids_full_path, metadata in scans.iterrows():
+        if metadata["modality_num"] != "20217" and metadata["modality_num"] != "20225":
+            copy_file_to_bids(
+                zipfile=str(dataset_directory) + "/" + metadata["source_zipfile"],
+                filenames=[metadata["source_filename"]] + metadata["sidecars"],
+                bids_path=to / bids_full_path,
+            )
+        else:
+            convert_dicom_to_nifti(
+                zipfiles=str(dataset_directory) + "/" + metadata["source_zipfile"],
+                bids_path=to / bids_full_path,
+            )
+            if metadata["modality_num"] == "20217":
+                import_event_tsv(bids_path=to / bids_full_path)
+    return
+
+
+def copy_file_to_bids(zipfile: str, filenames: List[str], bids_path: str) -> None:
+    """Install the requested files in the BIDS  dataset."""
+    import fsspec
+
+    fo = fsspec.open(zipfile)
+    fs = fsspec.filesystem("zip", fo=fo)
+    for filename in filenames:
+        if fs.exists(filename):
+            bids_path_extension = str(bids_path) + "." + (filename.split(".", 1)[1])
+            with fsspec.open(bids_path_extension, mode="wb") as f:
+                f.write(fs.cat(filename))
+
+
+def convert_dicom_to_nifti(zipfiles: str, bids_path: str) -> None:
+    """Install the requested files in the BIDS  dataset.
+    First, the dicom is extracted in a temporary directory
+    Second, the dicom extracted is converted in the right place using dcm2niix"""
+    import os
+    import subprocess
+    import tempfile
+    import zipfile
+    from pathlib import Path
+
+    zf = zipfile.ZipFile(zipfiles)
+    try:
+        os.makedirs(Path(bids_path).parent)
+    except OSError:
+        # Folder already created with previous instance
+        pass
+    with tempfile.TemporaryDirectory() as tempdir:
+        zf.extractall(tempdir)
+        command = [
+            "dcm2niix",
+            "-w",
+            "0",
+            "-f",
+            Path(bids_path).name,
+            "-o",
+            Path(bids_path).parent,
+        ]
+        command += ["-9", "-z", "y"]
+        command += ["-b", "y", "-ba", "y"]
+        command += [tempdir]
+        subprocess.run(command)
+    return
+
+
+def select_sessions(x: DataFrame) -> Series:
+    if x["source_sessions_number"] == "2":
+        return x.age_when_attended_assessment_centre_f21003_2_0
+    elif x["source_sessions_number"] == "3":
+        return x.age_when_attended_assessment_centre_f21003_3_0
+
+
+def import_event_tsv(bids_path: str) -> None:
+    """import the csv containing the events information."""
+    import os
+    from pathlib import Path
+
+    from fsspec.implementations.local import LocalFileSystem
+
+    fs = LocalFileSystem(auto_mkdir=True)
+    path_to_event_tsv = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))),
+        "resources",
+        "fmri",
+        "ukb_event.tsv",
+    )
```

Comment:
I wonder whether it would not make sense to just store this file under something like `resources/ukb/task-task-facesshapesemotion_events.tsv` and copy it as is to the output BIDS directory.

That would avoid the extra code for renaming and computing the relative path with `.parent` which looks a bit weird to me.